OutOfSchool/OutOfSchool.DataAccess/Repository/EntityRepositorySoftDeleted.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Security.AccessControl;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using OutOfSchool.Services.Enums;
+using OutOfSchool.Services.Extensions;
+using OutOfSchool.Services.Models;
+
+namespace OutOfSchool.Services.Repository;
+
+public class EntityRepositorySoftDeleted<TKey, TEntity> : EntityRepositoryBase<TKey, TEntity>, IEntityRepositorySoftDeleted<TKey, TEntity>
+    where TEntity : class, IKeyedEntity<TKey>, ISoftDeleted, new()
+    where TKey : IEquatable<TKey>
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="EntityRepositorySoftDeleted{TKey, TEntity}"/> class.
+    /// </summary>
+    /// <param name="dbContext">OutOfSchoolDbContext.</param>
+    public EntityRepositorySoftDeleted(OutOfSchoolDbContext dbContext)
+        : base(dbContext)
+    {
+    }
+
+    /// <inheritdoc/>
+    public override async Task<IEnumerable<TEntity>> GetAll()
+    {
+        return await dbSet.Where(x => !x.IsDeleted).ToListAsync().ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
+    public override async Task<IEnumerable<TEntity>> GetAllWithDetails(string includeProperties = "")
+    {
+        IQueryable<TEntity> query = dbSet.Where(x => !x.IsDeleted);
+
+        foreach (var includeProperty in includeProperties.Split(
+                     new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries))
+        {
+            query = query.Include(includeProperty);
+        }
+
+        return await query.ToListAsync().ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
+    public override async Task<IEnumerable<TEntity>> GetByFilter(
+        Expression<Func<TEntity, bool>> predicate,
+        string includeProperties = "")
+    {
+        predicate = GetWhereExpression(predicate);
+        return await base.GetByFilter(predicate, includeProperties).ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
+    public override IQueryable<TEntity> GetByFilterNoTracking(
+        Expression<Func<TEntity, bool>> predicate,
+        string includeProperties = "")
+    {
+        predicate = GetWhereExpression(predicate);
+        return base.GetByFilterNoTracking(predicate, includeProperties);
+    }
+
+    /// <inheritdoc/>
+    public override Task<TEntity> GetById(TKey id) => dbSet.FirstOrDefaultAsync(x => !x.IsDeleted && x.Id.Equals(id));
+
+    /// <inheritdoc/>
+    public override Task<int> Count(Expression<Func<TEntity, bool>> where = null)
```

Comment:
(Code style - not insisting, just general recommendation) Please try to avoid usage of keywords as identifiers. 
Argument `where` here and in subsequent methods could be renamed to `condition` / `whereExpression` / other variant.

Reply:
Ok. Done.

Reply:
Looks like it was present in too many places. Thanks for thorough work!